python/ray/serve/_private/http_proxy.py

```diff
@@ -106,14 +108,9 @@ async def _handle_streaming_response(
             for asgi_message in asgi_messages:
                 # There must be exactly one "http.response.start" message that
                 # always contains the "status" field.
+                # TODO: fix.
                 if not status_code:
-                    assert asgi_message["type"] == "http.response.start", (
-                        "First response message must be 'http.response.start'",
-                    )
-                    assert "status" in asgi_message, (
-                        "'http.response.start' message must contain 'status'",
-                    )
-                    status_code = str(asgi_message["status"])
+                    status_code = str(asgi_message.get("status", "UNKNOWN"))
```

Comment:
Why don't we raise an `AssertionError` based on `type` or `status` anymore?

Reply:
because we now support websockets which have a different type and do not have a status

Reply:
updated the comment and explicitly checking message type now